src/main/scala/com/trueaccord/scalapb/json/JsonFormat.scala

```diff
@@ -351,14 +352,17 @@ object JsonFormat {
     }
   }
 
-  def parsePrimitiveByScalaType(scalaType: ScalaType, value: JValue, onError: => PValue): PValue = (scalaType, value) match {
+  def parsePrimitive(scalaType: ScalaType, protoType: FieldDescriptorProto.Type, value: JValue, onError: => PValue): PValue = (scalaType, value) match {
     case (ScalaType.Int, JInt(x)) => PInt(x.intValue)
     case (ScalaType.Int, JDouble(x)) => PInt(x.intValue)
     case (ScalaType.Int, JDecimal(x)) => PInt(x.intValue)
+    case (ScalaType.Int, JString(x)) if protoType == FieldDescriptorProto.Type.TYPE_INT32 => parseInt32(x)
```

Comment:
What about SINT32 and SFIXED32? Also, please use protoType.isSint32 for enum comparison, since `==` is not type safe.

Reply:
Fixed